Fixes InvGroup Form submission error and TemplateList Missing Resource Bug

AlexSCorey opened the pull request:
##### SUMMARY
This addresses #5736 and #5727 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.1
```


##### ADDITIONAL INFORMATION
![InvGroupForm](https://user-images.githubusercontent.com/39280967/72926541-9cf97d00-3d22-11ea-83ed-4dc78613cc8d.gif)

AlexSCorey commented on awx/ui_next/src/screens/Template/TemplateList/TemplateListItem.jsx:
```diff
@@ -58,101 +58,93 @@ const RightActionButtonCell = styled(ActionButtonCell)`
-    const { i18n, template, isSelected, onSelect } = this.props;
-    const canLaunch = template.summary_fields.user_capabilities.start;
-    const missingResourceIcon =
+function TemplateListItem({ i18n, template, isSelected, onSelect }) {
```
Most of this file is different simply because I converted it to a functional component.  The real changes are on lines 63-65.

jakemcdermott reviewed (changes requested):
I like what you've done so far. These are good bug fixes and you've shown some great initiative by updating the `TemplateListItem` to a functional component. Nice job there. :+1: 

That said, two bugfixes and a component refactor is a little much for one commit. Please separate this into three commits - one for each bug and a third for the refactor. Please also make sure that the tests pass locally after adding each commit.

AlexSCorey committed:
Fixes InvGroupForm submission bug.

The inventory id now comes through useParams instead of through props.
Also updates tests to reflect those changes

AlexSCorey committed:
Ensures no missingResourceIcon for WFJT on JTList

Adds a test to ensure that missingResourceIcon is not shown for WFJTs.

AlexSCorey committed:
Moves TemplateListItem to a functional component
